packages/mui-utils/src/deepmerge.ts

```diff
@@ -6,15 +6,20 @@ export interface DeepmergeOptions {
   clone?: boolean;
 }
 
-function deepClone<T>(source: T): T | Record<keyof any, unknown> {
+function deepClone<T>(source: T, visited = new Set()): T | Record<keyof any, unknown> {
   if (!isPlainObject(source)) {
     return source;
   }
 
   const output: Record<keyof any, unknown> = {};
 
   Object.keys(source).forEach((key) => {
-    output[key] = deepClone(source[key]);
+    if (!visited.has(source[key])) {
```

Comment:
I get that this fix is not really cloning circular objects, but I can't think of a better way to handle circular objects